### PR TITLE
feat: add divider component

### LINKED ISSUE #179: Add Divider component
### Expected behavior  
Starting point https://fluentweb.com/components/divider but, we will re-examine from the ground up what expect requirements we want to include.

- Ensure style is maintained
- Ensure behavior is maintained
- Ensure accessibility is maintained

For testing follow https://github.com/Microsoft/fast-dna/wiki/the-process

### PR DESCRIPTION
**Pull request by awentzel**
<body>

<footer>
 
closes #179

### CODE REVIEW

File: packages/fast-components-react-base/src/divider/divider.props.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * Divider HTML Roles
+ */
+export enum DividerRole {
+    presentation = 'presentation',
+    separator = 'separator'
+}
+
+export interface IDividerProps {
```
Comment (nicholasrice) [inline packages/fast-components-react-base/src/divider/divider.props.ts]: I proposed a new naming convention for props interfaces in my snapshot PR - this interface is representative of "handled" props so I think it should be named accordingly. Take a look at https://github.com/Microsoft/fast-dna/pull/207 in the button.props.ts file for details.

File: packages/fast-components-react-base/src/divider/divider.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    protected handledProps: HandledProps<IDividerProps & IManagedClasses<IDividerClassNameContract>> = {
+        managedClasses: void 0,
+        role: void 0
+    };
+
+    /**
+     * Define default props
+     */
+    protected defaultProps: IDividerProps = {
+        role: void(0)
```
Comment (nicholasrice) [inline packages/fast-components-react-base/src/divider/divider.tsx]: You don't need `defaultProps` in this instance - we don't need to set all defaultProps to `void(0)` anymore. Or, you could also default to `DividerRole.separater` if that is the logical default.

File: packages/fast-components-styles-msft/src/divider/index.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import {IDesignSystem} from "../design-system";
+import {ComponentStyles} from "@microsoft/fast-jss-manager";
+import {IDividerClassNameContract} from "@microsoft/fast-components-class-name-contracts";
+
+const styles: ComponentStyles<IDividerClassNameContract, IDesignSystem> = {
+    divider: {
+
```
Comment (nicholasrice) [inline packages/fast-components-styles-msft/src/divider/index.ts]: we don't add vertical spacing or anything to divider?

File: packages/fast-components-react-base/src/divider/divider.tsx (hunk 2 above)
Comment (awentzel) [reply packages/fast-components-react-base/src/divider/divider.tsx]: DividerRole.separator is intrinsically set as part of HTML5 so there default is null.  I can remove. I also, do like the new name.

File: packages/fast-components-react-base/src/divider/divider.props.ts (hunk 1 above)
Comment (awentzel) [reply packages/fast-components-react-base/src/divider/divider.props.ts]: this makes sense for better readability.

File: packages/fast-components-styles-msft/src/divider/index.ts (hunk 3 above)
Comment (awentzel) [reply packages/fast-components-styles-msft/src/divider/index.ts]: @chrisdholt  and I had discussed this and think the spacing helpers of the past are not needed at this time.  Divider, Image, Card were using it.  We should revisit spacing as part of a cross-cutting feature probably and figure out how to build that into the design system if its needed.  Chris may have some additional thoughts here as well.

File: packages/fast-components-class-name-contracts/src/index.ts [hunk 4]
```diff
@@ -1,2 +1,3 @@
+export * from "./divider";
```
Comment (chrisdholt) [inline packages/fast-components-class-name-contracts/src/index.ts]: should be alphabetical

File: packages/fast-components-react-base/src/divider/divider.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, {HandledProps} from "../foundation";
+import {DividerRole, IDividerHandledProps} from './divider.props';
+import {IDividerClassNameContract, IManagedClasses} from "@microsoft/fast-components-class-name-contracts";
+
+class Divider extends Foundation<IDividerHandledProps & IManagedClasses<IDividerClassNameContract>, React.AllHTMLAttributes<HTMLElement>, {}> {
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/divider/divider.tsx]: I think you want: 
`
React.HTMLAttributes<HTMLHRElement>
`

File: packages/fast-components-react-base/src/divider/docs.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+## Best practices
+Use the *divider* to show a change in the content between two blocks of information. The content surrounding the *divider* determine the spacing above and below it. For example, headings of differing importance require different spacing above them, but it is possible to apply spacing preselects for space above and/or below the *divider*.
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/divider/docs.md]: preselects?

File: packages/fast-components-react-base/src/divider/docs.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+### Usage guidance
+Use *divider* to separate sections that may not otherwise have a clear beginning or end, and to separate information blocks where the context or continuity changes between the blocks.
+
+### Style guidance
+When white space or headings can clearly define new sections, dividers are not needed. They are also not for graphic decoration.
+
+## Accessibility
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/divider/docs.md]: I feel like the Accessibility stuff is really all that makes sense in the base here.

File: packages/fast-components-react-base/src/index.ts [hunk 8]
```diff
@@ -1,3 +1,7 @@
+import Divider from "./divider";
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/index.ts]: alphabetize

File: packages/fast-components-react-msft/src/divider/examples.data.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import {IGenericExample} from "../examples";
+import {IManagedClasses} from "@microsoft/fast-jss-manager-react";
+import Divider from "./index";
+import {DividerRole, IDividerHandledProps} from "@microsoft/fast-components-react-base";
+
+export default {
+    name: "divider",
+    component: Divider,
+    data: [
+        {
```
Comment (chrisdholt) [inline packages/fast-components-react-msft/src/divider/examples.data.ts]: can you try using:

`role: void 0`

An empty object seems like a mistake, but I understand what you are doing

File: packages/fast-components-react-msft/src/divider/index.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import * as React from "react";
+import { Divider, IDividerClassNameContract, IDividerHandledProps, IFoundationProps } from "@microsoft/fast-components-react-base";
```
Comment (chrisdholt) [inline packages/fast-components-react-msft/src/divider/index.ts]: can you pull the spaces between the brackets and the text per our most current conversation for consistency?

File: packages/fast-components-styles-msft/src/divider/index.ts (hunk 3 above)
Comment (chrisdholt) [reply packages/fast-components-styles-msft/src/divider/index.ts]: @awentzel I think we'll want a default here for vertical spacing and we may want to include any of the styles pertaining to a css reset or at least ensure that everything works across all browsers:

`
box-sizing: content-box;
height: 0;
margin-top: 12px;
margin-bottom: 12px;
border-top: 1px solid $color-foreground-20;
`

We can come back in and setup that spacing to be dynamic or based on a consistent utility

File: packages/fast-components-styles-msft/src/index.ts [hunk 11]
```diff
@@ -1,3 +1,9 @@
+/**
+ * Export component styles
+ */
+import DividerStyles from "./divider";
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/index.ts]: alphabetize

File: packages/fast-components-class-name-contracts/src/index.ts (hunk 4 above)
Comment (awentzel) [reply packages/fast-components-class-name-contracts/src/index.ts]: @chrisdholt could you file a new issue to enforce this in fast-tslint-rules, I think @nicholasrice makes a good point.

File: packages/fast-components-react-base/src/divider/docs.md (hunk 6 above)
Comment (awentzel) [reply packages/fast-components-react-base/src/divider/docs.md]: Updated

File: packages/fast-components-class-name-contracts/src/index.ts (hunk 4 above)
Comment (chrisdholt) [reply packages/fast-components-class-name-contracts/src/index.ts]: Still not sure it's a nit, although I agree that tslint should catch this. More than happy to discuss various options with the team. We should have some kind of order though aside from random import/exportss in my opinion.

File: packages/fast-components-react-base/src/divider/examples.data.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import {ICategoryItemProps} from "@microsoft/fast-development-site-react";
```
[the remark this answers is not in the document]
Comment (chrisdholt) [reply packages/fast-components-react-base/src/divider/examples.data.ts]: Yes, I totally agree. We should name these accordingly as test fixtures.

**Pull request merged**